tests/test_photo.py

```diff
@@ -1,14 +1,65 @@
-import pytest, requests
-from typing import Generator
+import pytest
+import requests
+import os
+from pathlib import Path
+from PIL import Image
 
+photo_process_host = "http://localhost:85"
 
-photo_host = "http://localhost:5000"
-filepath = "./tests/resources/test.jpg"
+parent_path = Path(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
+image_path = os.path.join(str(parent_path) + "/tests/resources/test.jpg")
+storage_path = os.path.join(str(parent_path) + "/container-storage")
+image_id = "101"
 
-def test_put_photo():
-    test_file = ("test.jpg", open(filepath, "rb"), "image/jpeg")
+
+@pytest.fixture
+def set_photo():
     response = requests.put(
-        url="{0}/photo/101".format(photo_host), files={"file": test_file}
-    ).json()
-    print(response)
+        url="{0}/photo/{1}".format(photo_process_host, image_id),
+        files={"file": ("test.jpg", open(image_path, "rb"), "image/jpeg")},
+    )
+    return response
+
+
+def test_put_photo(set_photo):
+    assert set_photo.status_code == 200
+    image_storage_path = Path(os.path.join(storage_path, "{}.jpg".format(image_id)))
+    assert image_storage_path.exists()
+
+    # cleanup
+    image_storage_path.unlink()
+
+
+def test_get_photo_and_similar(set_photo):
+    # get photo resized to 100
+    response = requests.get(
+        url="{0}/photo/{1}".format(photo_process_host, image_id), data={"resize": 100}
+    )
+    assert response.status_code == 200
+    temp_image_path = os.path.join(str(parent_path) + "/tests/resources/temp.jpg")
+
+    # store the resized photo
+    with open(temp_image_path, "wb") as f:
+        f.write(response.content)
+
+    im = Image.open(temp_image_path)
+    assert im.format == "JPEG"
+
+    response = requests.get(
+        url="{0}/photo/similar".format(photo_process_host, image_id),
+        files={"file": ("temp.jpg", open(temp_image_path, "rb"), "image/jpeg")},
+    )
+    response.status_code == 200
```

Comment:
For test_get_photo_and_similar function maybe you could check the response too?

Reply:
check the response like this?
```python
    with open(temp_image_path, "wb") as f:
        f.write(response.content)

    im = Image.open(temp_image_path)
    assert im.format == "JPEG"
```

Reply:
No, i was thinking about this request 
    response = requests.get(
        url="{0}/photo/similar".format(photo_process_host, image_id),
        files={"file": ("temp.jpg", open(temp_image_path, "rb"), "image/jpeg")},
    )

It returns a json with similar images, which could be tested
1. if it can be parsed as a json
2. if t contains all fields required

Reply:
got it.

Reply:
```python
    response = requests.put(
        url="{0}/photo/similar".format(photo_process_host),
        files={"file": ("temp.jpg", open(temp_image_path, "rb"), "image/jpeg")},
    )
    assert response.status_code == 200
    assert response.json() == [101]
```

Is this good?

Reply:
Yep